'grunt eslint' passes; `grunt eslint:tests` fails; break git pre-push hook

I had pushed to remote successfully just a moment ago, but after pulling from mozilla/master and picking up "nothing" now my git push fails, complaining about two files I haven't changed. Oddly, `grunt lint` run directly passes with no errors.

Also, `eslint` is complaining about using `self` not `this`, but in this context `self` is an argument to a function, so `eslint` apparently doesn't have a clue what it is whining about.

```
jrgm@ubuntu:~/fxa-content-server$ ./node_modules/.bin/grunt lint
Running "eslint:grunt" (eslint) task


Running "eslint:app" (eslint) task


Running "eslint:tests" (eslint) task


Running "eslint:scripts" (eslint) task


Running "eslint:server" (eslint) task


Running "jsonlint:app" (jsonlint) task
>> 3 files lint free.

Running "jscs:server" (jscs) task
>> 149 files without code style errors.

Running "jscs:app" (jscs) task
>> 271 files without code style errors.

Running "amdcheck:app" (amdcheck) task

Total unused paths: 0 in 0 files.
Total unused dependencies: 0 in 0 files.
Total processed files: 335

Done, without errors.


Execution Time (2015-07-09 01:12:30 UTC)
loading tasks   1.2s  ▇▇▇▇ 8%
eslint:grunt   872ms  ▇▇▇ 5%
eslint:app        6s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 37%
eslint:tests    1.5s  ▇▇▇▇▇ 9%
eslint:server  349ms  ▇▇ 2%
jsonlint:app   502ms  ▇▇ 3%
jscs:server    602ms  ▇▇ 4%
jscs:app        2.4s  ▇▇▇▇▇▇▇ 15%
amdcheck:app    2.7s  ▇▇▇▇▇▇▇▇ 17%
Total 16.2s

jrgm@ubuntu:~/fxa-content-server$ git push jrgm fix-metrics-test-value
Running "newer:eslint" (newer) task

Running "newer:eslint:grunt" (newer) task

Running "eslint:grunt" (eslint) task


Running "newer-postrun:eslint:grunt:1:/home/jrgm/fxa-content-server/node_modules/grunt-newer/.cache" (newer-postrun) task

Running "newer:eslint:app" (newer) task
No newer files to process.

Running "newer:eslint:tests" (newer) task

Running "eslint:tests" (eslint) task

tests/functional/force_auth.js
  23:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this
  32:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this

tests/functional/oauth_force_email.js
  27:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this

✖ 3 problems (3 errors, 0 warnings)

Warning: Task "eslint:tests" failed. Use --force to continue.

Aborted due to warnings.


Execution Time (2015-07-09 01:12:52 UTC)
loading tasks      1.3s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 30%
eslint:grunt      853ms  ▇▇▇▇▇▇▇▇▇ 20%
newer:eslint:app  389ms  ▇▇▇▇▇ 9%
eslint:tests       1.7s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 40%
Total 4.2s


husky - pre-push hook failed (add --no-verify to bypass)

error: failed to push some refs to 'git@github.com:jrgm/fxa-content-server.git'
```

```
jrgm@ubuntu:~/fxa-content-server$ ./node_modules/.bin/grunt eslint
Running "eslint:grunt" (eslint) task


Running "eslint:app" (eslint) task


Running "eslint:tests" (eslint) task


Running "eslint:scripts" (eslint) task


Running "eslint:server" (eslint) task


Done, without errors.


Execution Time (2015-07-09 02:12:30 UTC)
loading tasks   1.2s  ▇▇▇▇▇▇ 12%
eslint:grunt   883ms  ▇▇▇▇▇ 9%
eslint:app        6s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 59%
eslint:tests    1.7s  ▇▇▇▇▇▇▇▇ 17%
eslint:server  441ms  ▇▇▇ 4%
Total 10.3s
```
```
jrgm@ubuntu:~/fxa-content-server$ ./node_modules/.bin/grunt newer:eslint
Running "newer:eslint" (newer) task

Running "newer:eslint:grunt" (newer) task

Running "eslint:grunt" (eslint) task


Running "newer-postrun:eslint:grunt:1:/home/jrgm/fxa-content-server/node_modules/grunt-newer/.cache" (newer-postrun) task

Running "newer:eslint:app" (newer) task
No newer files to process.

Running "newer:eslint:tests" (newer) task

Running "eslint:tests" (eslint) task

tests/functional/force_auth.js
  23:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this
  32:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this

tests/functional/oauth_force_email.js
  27:2  error  Designated alias 'self' is not assigned to 'this'  consistent-this

✖ 3 problems (3 errors, 0 warnings)

Warning: Task "eslint:tests" failed. Use --force to continue.

Aborted due to warnings.


Execution Time (2015-07-09 02:12:48 UTC)
loading tasks      1.2s  ▇▇▇▇▇▇▇▇▇▇▇▇▇ 29%
eslint:grunt      910ms  ▇▇▇▇▇▇▇▇▇▇ 21%
newer:eslint:app  369ms  ▇▇▇▇ 9%
eslint:tests       1.7s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 40%
Total 4.2s

jrgm@ubuntu:~/fxa-content-server$
```

--- a/tests/functional/force_auth.js
+++ b/tests/functional/force_auth.js
@@ -20,8 +20,8 @@ define([
   var email;
   var client;
 
-  function openFxa(self, email) {
-    return self.remote
+  function openFxa(context, email) {
+    return context.remote
       .setFindTimeout(intern.config.pageLoadTimeout)
       .get(require.toUrl(FORCE_AUTH_URL + '?email=' + email))
 
@@ -29,8 +29,8 @@ define([
       .end();
   }
 
-  function attemptSignIn(self) {
-    return self.remote
+  function attemptSignIn(context) {
+    return context.remote
       // user should be at the force-auth screen
       .findByCssSelector('#fxa-force-auth-header')
       .end()

--- a/tests/functional/oauth_force_email.js
+++ b/tests/functional/oauth_force_email.js
@@ -24,8 +24,8 @@ define([
     return FunctionalHelpers.openFxaFromRp(context, 'force_auth', emailParam);
   }
 
-  function attemptSignIn(self) {
-    return self.remote
+  function attemptSignIn(context) {
+    return context.remote
       // user should be at the force-auth screen
       .findByCssSelector('#fxa-force-auth-header')
       .end()

